jetty-util/src/main/java/org/eclipse/jetty/util/statistic/SampleStatistic.java

```diff
@@ -55,61 +50,89 @@ public void reset()
         _totalVariance100.reset();
     }
 
-    public void set(final long sample)
+    /**
+     * Records a sample value.
+     *
+     * @param sample the value to record.
+     */
+    public void record(long sample)
     {
         long total = _total.addAndGet(sample);
         long count = _count.incrementAndGet();
 
-        if (count>1)
+        if (count > 1)
         {
-            long mean10 = total*10/count;
-            long delta10 = sample*10 - mean10;
-            _totalVariance100.add(delta10*delta10);
+            long mean10 = total * 10 / count;
+            long delta10 = sample * 10 - mean10;
+            _totalVariance100.add(delta10 * delta10);
         }
 
         _max.accumulate(sample);
     }
 
     /**
-     * @return the max value
+     * @deprecated use {@link #record(long)} instead
+     */
+    @Deprecated
+    public void set(long sample)
+    {
+        record(sample);
+    }
+
+    /**
+     * @return the max value of the recorded samples
      */
     public long getMax()
     {
         return _max.get();
     }
 
+    /**
+     * @return the sum of all the recorded samples
+     */
     public long getTotal()
     {
         return _total.get();
     }
 
+    /**
+     * @return the number of samples recorded
+     */
     public long getCount()
     {
         return _count.get();
     }
 
+    /**
+     * @return the average value of the samples recorded, or zero if there are no samples
+     */
     public double getMean()
     {
-        return (double)_total.get()/_count.get();
+        long count = getCount();
+        return count > 0 ? (double)_total.get() / _count.get() : 0.0D;
     }
 
+    /**
+     * @return the variance of the samples recorded, or zero if there are less than 2 samples
+     */
     public double getVariance()
     {
-        final long variance100 = _totalVariance100.sum();
-        final long count = _count.get();
-
-        return count>1?((double)variance100)/100.0/(count-1):0.0;
+        long variance100 = _totalVariance100.sum();
+        long count = getCount();
+        return count > 1 ? variance100 / 100.0D / (count - 1) : 0.0D;
     }
 
+    /**
+     * @return the standard deviation of the samples recorded
+     */
     public double getStdDev()
     {
         return Math.sqrt(getVariance());
     }
 
-    /* ------------------------------------------------------------ */
     @Override
     public String toString()
     {
-        return String.format("%s@%x{c=%d,m=%d,t=%d,v100=%d}",this.getClass().getSimpleName(),hashCode(),_count.get(),_max.get(),_total.get(),_totalVariance100.sum());
+        return String.format("%s@%x{c=%d,m=%d,t=%d,v100=%d}", getClass().getSimpleName(), hashCode(), getCount(), getMax(), getTotal(), _totalVariance100.sum());
```

Comment:
We may as well report the actual stddev here rather than the v100 (which is even more meaningless than stddev).
I guess there is no cheap way of calculating the median on the fly?